### ISSUE TITLE
Recurring-retention sentinel probes score disc_r1/r2/r3/disc_r1_pscale with a forward pass they never deploy (flag_hidden_rms bypasses the hidden-RMS guard)

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#319]
## Summary

`run_recurring_ipmnist_retention_development` (`alberta_framework/benchmarks/ipmnist_screening.py`) scores every sentinel probe with `mlp_logits(params, spec.frozen_probe_input(state, sentinel_inputs, hp))`. `_hidden_rms_frozen_probe_input` exists precisely to refuse this for arms whose deployed forward pass RMS-normalizes the hidden layers — its docstring: *"the probe harness computes logits with `mlp_logits`, so any input-side transform would silently probe the wrong model. Failing closed here is the honest option."* It is wired to `sigma0_hidden_norm` (hyperparameter key `hidden_rms`).

The discovered-rule family switches on the same hidden-RMS forward pass through a different key, `flag_hidden_rms` (`_make_discovered_rule_learner`: `_rms_loss` / `_loss_with_hidden` normalize both ReLU layers). `disc_r1`, `disc_r2`, `disc_r3` (all `flag_hidden_rms=1.0`) and `disc_r1_pscale` (`flag_hidden_rms=1.0`) are nevertheless registered with `frozen_probe_input=_ema_frozen_probe_input`, so the recurring-retention harness scores the plain MLP on those arms' parameters — a model they never run.

The online accuracy trace is unaffected (it comes from the arm's own step function). Everything the retention lane exists to measure is: `acquisition_end / pre_revisit / revisit_end` sentinel accuracy and hence `peak_to_revisit_forgetting`, `retention_change_from_acquisition`, and `revisit_recovery`.

## Falsifier (deterministic, pristine `main` @ `bae17ee`)

Registry scan:

```python
import alberta_framework.benchmarks.ipmnist_screening as S
for name, spec in S.SCREENING_REGISTRY.items():
    hp = spec.hyperparameters
    if hp.get("hidden_rms", 0.0) or hp.get("flag_hidden_rms", 0.0):
        print(name, spec.frozen_probe_input.__name__)
```

```
sigma0_hidden_norm _hidden_rms_frozen_probe_input
disc_r1 _ema_frozen_probe_input
disc_r2 _ema_frozen_probe_input
disc_r3 _ema_frozen_probe_input
disc_r1_pscale _ema_frozen_probe_input
```

End-to-end on a 16-dim / 4-class synthetic stream (`n_tasks=3, task_length=400, hidden 64/32`, phases `(400,400,400)`, 100 sentinels, seed 11) — reported sentinel accuracy from the harness vs. the same frozen `params` scored with the arm's actual hidden-RMS forward pass:

```
=== disc_r1_pscale (flag_hidden_rms=1.0) ===
  phase perm  reported(mlp_logits)  deployed(hidden-RMS forward)
    0    A      0.8200                 0.7600   delta=-0.0600
    1    A      0.3200                 0.3400   delta=+0.0200
    1    B      0.7500                 0.8100   delta=+0.0600
    2    A      0.8200                 0.8000   delta=-0.0200
  REPORTED : acq=0.8200 pre_revisit=0.3200 revisit_end=0.8200 forgetting=0.5000 recovery=0.5000
  DEPLOYED : acq=0.7600 pre_revisit=0.3400 revisit_end=0.8000 forgetting=0.4200 recovery=+0.4600

=== disc_r1_pscale_norms (flag_hidden_rms=0.0, same factory)  <- control
    every delta=+0.0000, REPORTED == DEPLOYED
```

The `flag_hidden_rms=0.0` sibling from the same factory shows the discrepancy is caused by the flag, not by noise. `disc_r1` shows the same pattern (forgetting 0.48 reported vs 0.45 deployed).

Expected: no arm whose forward pass diverges from `mlp_logits` can be registered with an input-side probe. On `main` that means routing the four `flag_hidden_rms` arms to the fail-closed `_hidden_rms_frozen_probe_input`, exactly like `sigma0_hidden_norm`.

## Relation to #190

The draft #190 replaces the fail-closed probe with `_hidden_rms_frozen_probe_logits` (keyed on `hidden_rms_epsilon`) and rewires `sigma0_hidden_norm`, `rff_rls`, `lin_rls`, `naive_bayes`, `nb_ensemble`, `rls_head` — but its diff has zero hits for `flag_hidden_rms` / `disc_r`, so the four discovered-rule arms keep `frozen_probe_logits=None` and still fall back to `mlp_logits` after #190. Whichever lands, these arms need the discovered-rule forward (`_hidden_rms_normalize` on both hidden layers with `hp["hidden_rms_epsilon"]`) or a fail-closed refusal.

## Dedup

`gh issue/pr list --search` for "hidden_rms", "flag_hidden_rms", "frozen_probe", "sentinel probe", "disc_r1", "recurring retention": only #18/#46 (sigma0 reduction pins, closed) and #190 (above). No issue covers this.

Fix PR follows (failing-test-first; fail-closed on `main`, composable with #190's direction).
